### ISSUE TITLE
App built in release configuration fails to launch with PostgreSQL database

### ISSUE THREAD

**Issue by cjmao** [vapor/postgresql#28]
### Steps to reproduce

``` bash
$ vapor new VaporBeta --branch=beta
```

Then, change the default sqlite database to postgres database:
``` diff
diff --git a/Package.resolved b/Package.resolved
index b6b2f4e..a3fbbed 100644
--- a/Package.resolved
+++ b/Package.resolved
......

diff --git a/Package.swift b/Package.swift
index f3127d7..24944ac 100644
--- a/Package.swift
+++ b/Package.swift
@@ -1,4 +1,4 @@
-// swift-tools-version:4.0
+// swift-tools-version:4.1
 import PackageDescription
 
 let package = Package(
@@ -7,11 +7,11 @@ let package = Package(
         // 💧 A server-side Swift web framework.
         .package(url: "https://github.com/vapor/vapor.git", from: "3.0.0-rc"),
 
-        // 🔵 Swift ORM (queries, models, relations, etc) built on SQLite 3.
-        .package(url: "https://github.com/vapor/fluent-sqlite.git", from: "3.0.0-rc"),
+        // 🔵 Swift ORM (queries, models, relations, etc).
+        .package(url: "https://github.com/vapor/fluent-postgresql.git", from: "1.0.0-rc"),
     ],
     targets: [
-        .target(name: "App", dependencies: ["FluentSQLite", "Vapor"]),
+        .target(name: "App", dependencies: ["FluentPostgreSQL", "Vapor"]),
         .target(name: "Run", dependencies: ["App"]),
         .testTarget(name: "AppTests", dependencies: ["App"]),
     ]
diff --git a/Sources/App/Models/Todo.swift b/Sources/App/Models/Todo.swift
index 5c77e09..bb03d35 100644
--- a/Sources/App/Models/Todo.swift
+++ b/Sources/App/Models/Todo.swift
@@ -1,8 +1,8 @@
-import FluentSQLite
+import FluentPostgreSQL
 import Vapor
 
 /// A single entry of a Todo list.
-final class Todo: SQLiteModel {
+final class Todo: PostgreSQLModel {
     /// The unique identifier for this `Todo`.
     var id: Int?
 
diff --git a/Sources/App/configure.swift b/Sources/App/configure.swift
index f09236f..a7e48b3 100644
--- a/Sources/App/configure.swift
+++ b/Sources/App/configure.swift
@@ -1,4 +1,4 @@
-import FluentSQLite
+import FluentPostgreSQL
 import Vapor
 
 /// Called before your application initializes.
@@ -10,7 +10,7 @@ public func configure(
     _ services: inout Services
 ) throws {
     // Register providers first
-    try services.register(FluentSQLiteProvider())
+    try services.register(FluentPostgreSQLProvider())
 
     // Register routes to the router
     let router = EngineRouter.default()
@@ -24,14 +24,14 @@ public func configure(
     middlewares.use(ErrorMiddleware.self) // Catches errors and converts to HTTP response
     services.register(middlewares)
 
-    // Configure a SQLite database
+    // Configure a PostgreSQL database
     var databases = DatabaseConfig()
-    try databases.add(database: SQLiteDatabase(storage: .memory), as: .sqlite)
+    try databases.add(database: PostgreSQLDatabase(config: .default()), as: .psql)
     services.register(databases)
 
     // Configure migrations
     var migrations = MigrationConfig()
-    migrations.add(model: Todo.self, database: .sqlite)
+    migrations.add(model: Todo.self, database: .psql)
     services.register(migrations)
 
     // Configure the rest of your application here

```

And last,
``` bash
$ vapor build --release
$ vapor run --release
```

### Expected behavior

The app should start running.

### Actual behavior

``` bash
Running VaporBeta ...
Migrating psql DB
[PostgreSQL] dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "Cannot initialize PostgreSQLAuthenticationType from invalid Int32 value 8", underlyingError: nil))
```

And the app hangs there without starting nor exiting.

### Environment

* Vapor Framework version: `3.0.0-rc.1.1`
* Vapor Toolbox version: `3.1.4`
* OS version: `macOS 10.13`, `Ubuntu 16.04`

**Comment by cjmao** [comment]
I tried dev builds both on my mac and on my ubuntu server, and everything is fine.

I think this might be something related to the swift compiler's optimization.
I tried to set the optimization level to none in Xcode's release configuration, and the app launched without this error.
Later, I tried to set the optimization level to `Optimize for Speed` and then `Optimize for Size` in debug configuration, this error also showed.

**Comment by cjmao** [comment]
Btw, I've pushed my code [here](https://github.com/cjmao/VaporBeta).
It contains a single commit with changes necessary to reproduce this issue.

**Comment by SandorDobi** [comment]
The NIO branch solved tihs problem. 
You need modify the package.swift to use NIO branch

```
.package(url: "https://github.com/vapor/vapor.git", .branch("nio"),
 .package(url: "https://github.com/vapor/fluent-postgresql", .branch("nio"))
```

**Comment by cjmao** [comment]
Fixed in `1.0.0-rc.2`.
